Implement CRSF teleop (#1)
husarion/husarion_ugv_crsf_teleop: README.md

# husarion_ugv_crsf_teleop

> This ROS 2 package allows you to control robots using a CRSF compatible remote control. Receiver should be connected to the robot's computer via USB-UART converter or be integrated as a hardware USB dongle. CRSF protocol parser is implemented based on the following [specification](https://github.com/crsf-wg/crsf/wiki).

Reviewer: ```suggestion
This ROS 2 package allows you to control robots using a CRSF compatible remote control. A receiver should be connected to the robot's computer via USB-UART converter or be integrated as a hardware USB dongle. The CRSF protocol parser is implemented based on the following [specification](https://github.com/crsf-wg/crsf/wiki).
```

## Launch Files

- `teleop.launch.py`: Launches crsf_teleop_node node. Automatically respawns node if it exits. Node's namespace can be set using the `namespace` launch argument.

## Configuration Files

> - [`crsf_teleop.yaml`](./config/crsf_teleop.yaml): Sets default parameter values for the `teleop.launch.py` launch file.

Reviewer: ```suggestion
- [`crsf_teleop.yaml`](./config/crsf_teleop.yaml): Sets default parameter values for the crsf_teleop_node when `teleop.launch.py` is launched.
```

## ROS Nodes

### crsf_teleop_node

Translates received CRSF commands to velocity commands for the robot.

The following channels are used for controlling the robot via the TX16S remote control:
- Channel 2 - Right gimbal up/down - forward/backward velocity
- Channel 4 - Left gimbal left/right - turning (angular) velocity
- Channel 5 - SF switch - emergency stop
- Channel 7 - SA switch (down position) - silence `cmd_vel` messages, allows other nodes to control the robot while enabling e_stop functionality
- Channel 11 - SG switch - tristate switch, selects robot speed

#### Publishes

- `/cmd_vel` [*geometry_msgs/Twist*]: Publishes velocity commands to the robot.

> - `/link_status` [*panther_crsf_teleop_msgs/LinkStatus*]: Describes radio link status between the remote control and the robot. Parameters are described in the [CRSF_FRAMETYPE_LINK_STATISTICS frame documentation](https://github.com/crsf-wg/crsf/wiki/CRSF_FRAMETYPE_LINK_STATISTICS).

Reviewer: ```suggestion
- `cmd_vel` [*geometry_msgs/Twist*]: Publishes velocity commands to the robot.
- `link_status` [*panther_crsf_teleop_msgs/LinkStatus*]: Describes radio link status between the remote control and the robot. Parameters are described in the [CRSF_FRAMETYPE_LINK_STATISTICS frame documentation](https://github.com/crsf-wg/crsf/wiki/CRSF_FRAMETYPE_LINK_STATISTICS).
```
